src/indexing.jl

```diff
@@ -176,9 +210,54 @@ Constructs a index for all the files located within  `dir`.
 Same as [`index_checkpoint_files`] except not restricted to files created by Checkpoints.jl.
 """
 function index_files(dir::AbstractPath)
+    depnames_lookup = _depnames_lookup()
     map(Iterators.filter(isfile, walkpath(dir))) do path
-        return IndexEntry(path, dir)
+        return IndexEntry(path, dir, depnames_lookup)
     end
 end
 
 index_files(dir) = index_files(Path(dir))
+
+"""
+    search(name::AbstractString, index)
```

Comment:
Is this part supposed to be in #46 instead?

Reply:
Yep, got my branches reversed. Updated both MRs to be the correct way round.